HIP/hip-690.md

```diff
@@ -33,22 +34,113 @@ Currently, each node has an address book configuration file which ties IP addres
+
+```
+/**
+* The data about a node, including its service endpoints and the Hedera account to be paid for
+* services provided by the node (that is, queries answered and transactions submitted.)
+*
+* If the <tt>serviceEndpoint</tt> list is not set, or empty, then the endpoint given by the
+* (deprecated) <tt>ipAddress</tt> and <tt>portno</tt> fields should be used.
+*
+* All fields are populated in the 0.0.102 address book file while only fields that start with # are
+* populated in the 0.0.101 address book file.
+*/
+  message NodeAddress {
+  /**
+  * The IP address of the Node with separator & octets encoded in UTF-8.  Usage is deprecated,
+  * ServiceEndpoint is preferred to retrieve a node's list of IP addresses and ports
+  */
+    bytes ipAddress = 1 [deprecated=true];
+
+  /**
+  * The port number of the grpc server for the node.  Usage is deprecated, ServiceEndpoint is
+  * preferred to retrieve a node's list of IP addresses and ports
+  */
+    int32 portno = 2 [deprecated=true];
+
+  /**
+  * Usage is deprecated, nodeAccountId is preferred to retrieve a node's account ID
+  */
+    bytes memo = 3 [deprecated=true];
+
+  /**
+  * The node's X509 RSA public key used to sign stream files (e.g., record stream
+  * files). Precisely, this field is a string of hexadecimal characters which,
+  * translated to binary, are the public key's DER encoding.  
+  */
+    string RSA_PubKey = 4;
+
+  /**
+  * # A non-sequential identifier for the node
+  */
+  int64 nodeId = 5;
+
+  /**
+  * # The account to be paid for queries and transactions sent to this node
+  */
+  AccountID nodeAccountId = 6;
+
+  /**
+  * # Hash of the node's TLS certificate. Precisely, this field is a string of
+  * hexadecimal characters which, translated to binary, are the SHA-384 hash of
+  * the UTF-8 NFKD encoding of the node's TLS cert in PEM format. Its value can be
+  * used to verify the node's certificate it presents during TLS negotiations.
+  */
+    bytes nodeCertHash = 7;
+
+  /**
+  * # A node's service IP addresses and ports
+  */
+  repeated ServiceEndpoint serviceEndpoint = 8;
+
+  /**
+  * A description of the node, with UTF-8 encoding up to 100 bytes
+  */
+    string description = 9;
+
+  /**
+  * [Deprecated] The amount of tinybars staked to the node
+  */
+    int64 stake = 10 [deprecated = true];
+  
+  /**
+    * The ownership type of the node. Whether the node belongs to a council member or a community member.
+    * If not present, defaults to 0 (COUNCIL).
+    */
+    NodeOwnershipType nodeOwnershipType = 0;
+  }
+  
+  /**
+  * Ownership type of the node. Whether the node belongs to a council member or a community member.
+  */
+  enum NodeOwnershipType {
+      COUNCIL = 0;
+      COMMUNITY = 1;
+  }
+```
+
+The mirrornode REST API will also expose this datum. The /network/nodes endpoint currently returns an array of `nodes` objects. 
+A new property `nodeOwnershipType` will be added to the `nodes` response object.
```

Comment:
The full API endpoint is `/api/v1/network/nodes`. The field would be in snake case and not have a redundant node prefix: `ownership_type`. We should show a simplified JSON example of the payload an mention the possibles values as string enums `"COUNCIL"` and `"COMMUNITY"`.

Reply:
Added